test/integration/MExtensionSystem.t.sol

```diff
@@ -0,0 +1,1548 @@
+
+        assertEq(aliceYieldWithFee, 11375, "alice's yield with fee should be 11375");
+        assertEq(aliceFee, 5687, "alice's fee should be 5687");
+        assertEq(aliceYield, 5688, "alice's yield should be 5688");
+
+        vm.prank(alice);
+        swapFacility.swap(address(mEarnerManager), address(wrappedM), mEarnerManagerBalance - 2, alice);
+
+        uint256 wrappedMBalance = wrappedM.balanceOf(alice);
+        assertEq(wrappedMBalance, 10e6 - 2);
+
+        vm.warp(vm.getBlockTimestamp() + 10 days);
+
+        vm.prank(alice);
+        swapFacility.swapOutM(address(wrappedM), wrappedMBalance, alice);
+
+        mEarnerManagerBalance = mEarnerManager.balanceOf(alice);
+
+        assertEq(mEarnerManagerBalance, 2);
+
+        mEarnerManager.claimFor(alice);
+
+        mEarnerManagerBalance = mEarnerManager.balanceOf(alice);
+
+        assertEq(mEarnerManagerBalance, 5696, "alice's claiming should have put her yield in her balance");
+
+        assertEq(
+            mEarnerManager.balanceOf(feeRecipient),
+            5693,
+            "Fee recipient should have fee claimed for on alice's claiming"
+        );
+
+        mYieldToOne.claimYield();
+
+        mYieldToOneBalance = mYieldToOne.balanceOf(yieldRecipient);
+
+        assertEq(mYieldToOneBalance, 11401, "yield recipient should have its yield claimed");
+
+        mYieldFee.claimYieldFor(alice);
+
+        mYieldFeeBalance = mYieldFee.balanceOf(alice);
+
+        assertEq(mYieldFeeBalance, 897145, "alice should have her yield claimed");
+    }
+
+    uint256 constant M_YIELD_TO_ONE = 0;
+    uint256 constant M_YIELD_FEE = 1;
+    uint256 constant M_EARNER_MANAGER = 2;
+
+    /// @dev Using lower fuzz runs and depth to avoid burning through RPC requests in CI
+    /// forge-config: default.fuzz.runs = 100
+    /// forge-config: default.fuzz.depth = 20
+    /// forge-config: ci.fuzz.runs = 10
+    /// forge-config: ci.fuzz.depth = 2
+    function testFuzz_yieldClaim_afterMultipleSwaps(uint256 seed) public {
+        vm.startPrank(alice);
+        mToken.approve(address(swapFacility), type(uint256).max);
+        mYieldFee.approve(address(swapFacility), type(uint256).max);
+        mYieldToOne.approve(address(swapFacility), type(uint256).max);
+        mEarnerManager.approve(address(swapFacility), type(uint256).max);
+        wrappedM.approve(address(swapFacility), type(uint256).max);
+        vm.stopPrank();
+
+        function(address, uint256[] memory, uint256) internal returns (uint256, uint256[] memory)[]
+            memory yieldAssertions = new function(address, uint256[] memory, uint256)
+                internal
+                returns (uint256, uint256[] memory)[](3);
+        yieldAssertions[M_YIELD_TO_ONE] = _testYieldCapture_mYieldToOne;
+        yieldAssertions[M_YIELD_FEE] = _testYieldCapture_mYieldFee;
+        yieldAssertions[M_EARNER_MANAGER] = _testYieldCapture_mEarnerManager;
```

Comment:
This kind of pattern is not easy to reason about.
I suppose it creates fuzz parameters that are passed to the various test functions.
But then why not just pass these to `testFuzz_yieldClaim_afterMultipleSwaps` and then call the test functions with those?

Reply:
What this does is allow the permutation this are invoked in to be fuzzed, where one fuzz run will have one order of invocation and another, another. 

To do that, the test function pointers themselves are put into an array, and are invoked in an order derived from the fuzz run's seed. 

The functions accept an array of uints as a parameter, which allows the test to accumulate values throughout their invocations which are used at the end to assert against the actual state of the extensions that were invoked in the fuzz run.